src/connectors/dynamic-widgets/connectDynamicWidgets.ts

```diff
@@ -170,8 +170,21 @@ const connectDynamicWidgets: DynamicWidgetsConnector =
           const attributesToRender =
             results.renderingContent?.facetOrdering?.facets?.order ?? [];
 
+          const transformedAttributesToRender = transformItems(
+            attributesToRender,
+            { results }
+          );
+
+          if (!Array.isArray(transformedAttributesToRender)) {
+            throw new Error(
+              withUsage(
+                'The `transformItems` option expects a function that returns an Array.'
+              )
+            );
+          }
```

Comment:
in the future this should be an invariant (like warning, but can have a slimmed down error message in production), but as we don't have that tooling yet, this is fine.

Reply:
For future reference, here's the `invariant` we use in React InstantSearch: https://github.com/algolia/react-instantsearch/blob/acf33c13d5b4fdcce66fb961fb3393d8bfdd658f/packages/react-instantsearch-hooks/src/utils/invariant.ts